src/__tests__/token-registry.e2e.test.ts

```diff
@@ -0,0 +1,295 @@
+import {
+  BaseTitleEscrowCommand,
+  TitleEscrowChangeHolderCommand,
+  TitleEscrowEndorseChangeOfOwnerCommand,
+  TitleEscrowNominateBeneficiaryCommand,
+} from "../commands/title-escrow/title-escrow-command.type";
+import { TokenRegistryIssueCommand } from "../commands/token-registry/token-registry-command.type";
+import { deployTokenRegistry } from "../implementations/deploy/token-registry";
+import { surrenderDocument } from "../implementations/title-escrow/surrenderDocument";
+import { issueToTokenRegistry } from "../implementations/token-registry/issue";
+import { ConnectedSigner, getWalletOrSigner } from "../implementations/utils/wallet";
+import { TransactionReceipt } from "@ethersproject/providers";
+import { acceptSurrendered } from "../implementations/title-escrow/acceptSurrendered";
+import { rejectSurrendered } from "../implementations/title-escrow/rejectSurrendered";
+import { changeHolderOfTitleEscrow } from "../implementations/title-escrow/changeHolder";
+
+import ganache from "ganache";
+import { endorseChangeOfOwner } from "../implementations/title-escrow/changeOwner";
+import { nominateBeneficiary } from "../implementations/title-escrow/nominateBeneficiary";
+import { endorseNominatedBeneficiary } from "../implementations/title-escrow/endorseNominatedBeneficiary";
+
+jest.mock("../implementations/utils/wallet", () => {
+  const originalModule = jest.requireActual("../implementations/utils/wallet");
+  return {
+    __esModule: true,
+    ...originalModule,
+    getWalletOrSigner: jest.fn(),
+  };
+});
+
+const accounts = {
+  mnemonic: "indicate swing place chair flight used hammer soon photo region volume shuffle",
+  owner: {
+    ethAddress: "0xe0A71284EF59483795053266CB796B65E48B5124",
+    publicKey: "0x02de2454a05cdb55780b85c04128233e31ac9179235607e4d6fa0c6b38140fb51a",
+    privateKey: "0xe82294532bcfcd8e0763ee5cef194f36f00396be59b94fb418f5f8d83140d9a7",
+  },
+  receiver: {
+    ethAddress: "0xcDFAcbb428DD30ddf6d99875dcad04CbEFcd6E60",
+    publicKey: "0x0396762cb3d373ddab0685bbd5e45ccaf7481d8deb5b75ab38704fba089abed629",
+    privateKey: "0xc58c1ff75001afdca8cecb61b47f36964febe4188b8f7b26252286ecae5a8879",
+  },
+};
+
+const regexEthAddress = new RegExp("^0x[a-fA-F0-9]{40}$");
+
+describe("token-registry", () => {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore mock static method
+  const mockedGetWalletOrSigner: jest.Mock = getWalletOrSigner as jest.Mock;
+
+  jest.setTimeout(90000);
+
+  let tokenRegistryAddress = "";
+  const defaultNetwork = "goerli";
+
+  const defaults = {
+    network: defaultNetwork,
+    gasPriceScale: 1,
+    dryRun: false,
+  };
+  const ganacheOptions = {
+    logging: {
+      debug: false,
+      quiet: true,
+      verbose: false,
+    },
+    chain: {
+      chainId: 3,
+    },
+    wallet: {
+      mnemonic: accounts.mnemonic,
+    },
+    fork: {
+      network: defaultNetwork,
+    },
+  } as any;
+
```

Comment:
do you want to shift constants out? then full uppercase them to signal the intent they probably will not change such as `DEFAULTS`,  `GANACHE_OPTIONS` etc.

also do shift helper functions out if possible, lets keep the tests less cluttered